connectors/sources/confluence.py

```diff
@@ -321,11 +321,9 @@ async def fetch_spaces(self):
 
     async def fetch_server_space_permission(self, url):
         try:
-            async for permissions in self.api_call(
-                url=os.path.join(self.host_url, url),
-            ):
-                permission = await permissions.json()
-                return permission
+            permissions = await self.api_call(url=os.path.join(self.host_url, url))
+            permission = await permissions.json()
```

Comment:
I noticed that you always do `response.json` - why not always return json from the api_call?

Reply:
We thought of doing that but we will have problem with the  method `download_attachment` as we are passing `api_call` to `generic_chunked_download_func`  Which requires full response instead of json. To avoid this, we can add new parameter `return_full_response` in `api_call` method, which when `true` would return response in place of response.json when we are calling it via "generic_chunked_download_func" method. Let us know if that sounds good to you.

Reply:
Ah I overlooked this - then makes sense that you return a response